File: Java/src/br/com/dominio/s06/Exe04.java
package br.com.dominio.s06;

import java.util.Scanner;

public class Exe04 {

	public static void main(String[] args) {
		// Variaveis
		float altura, peso_ideal;
		char sexo;
		Scanner teclado = new Scanner(System.in);
		
		//entradas
		System.out.println("Informe sua altura: ");
		altura = teclado.nextFloat();
		
		System.out.println("Informe o sexo m/f: ");
		sexo = teclado.next().charAt(0);
		
		// processamento
		if(sexo == 'm' || sexo == 'M') {
			peso_ideal = (float)(72.7 * altura) - 58; //cast
			System.out.printf("Seu peso ideal é %.2f ", peso_ideal);
		}
		if(sexo == 'f' || sexo == 'F') {
			peso_ideal = (float)(62.1 * altura) - 44.7f; //cast
			System.out.println("Seu peso ideal é " + peso_ideal);
		}
		if(sexo != 'm' && sexo != 'M' && sexo != 'f' && sexo != 'F') {
			System.out.println("Sexo não reconhecido! ");
			peso_ideal = 0;
		}
		teclado.close();
	}

}

File: Python/Seção06/Exe04.py
#entradas
sexo = input("Informe seu sexo: (M ou F)")
altura = float(input("Informe sua Altura: "))
#processamento
if sexo.lower() == 'm':
    peso_ideal = (72.7 * altura) - 58
elif sexo.lower() == 'f':
    peso_ideal = (62.1 * altura) - 44.7
else:
    peso_ideal = 0
    print("Sexo não reconhecido")
#saida
print("Seu peso ideal é {0:.3f}".format(peso_ideal))